File: Code/RenameStrategyIMGJPG.cs
using System.IO;
using System.Text.RegularExpressions;

namespace Code
{

    public class RenameStrategyIMGJPG : RenameStrategyBase
    {

        private const string mask = @"img_\d{8}_.*\.jpg";

        public override bool CanRename(string fullFileName)
        {
            var fileName = Path.GetFileName(fullFileName);
            if (!Regex.IsMatch(fileName, mask, RegexOptions.IgnoreCase)) return false;

            return true;
        }

        protected override string generateNewFileName(string current) => current.Substring(4);
    }
}

File: src/main/java/net/siudek/RenameStrategyImgJpg.java
package net.siudek;

import org.springframework.stereotype.Component;

@Component
class RenameStrategyImgJpg extends RenameStrategyBaseOnRegex {

  private static final String FILENAME_REGEX = "img_\\d{8}_.*\\.jpg";

  public RenameStrategyImgJpg() {
    super(FILENAME_REGEX);
  }

  @Override
  protected String generateNewFileName(String current) {
    return current.substring(4);
  }

}
